src/Components/pages/Play.jsx

```diff
@@ -107,35 +110,36 @@ class Play extends Component {
         });
     }
 
-    addUser(nickname,uniqueId) {
-        var nickname = "singled out user"
-        const {gameId} = this.state;
+    addUser(starting_money) {
+        const {gameId,nickname} = this.state;
         const gameRef = db.collection('Rooms').doc(gameId);
-        const userRef = gameRef.collection('users').doc(uniqueId);
+        const userRef = gameRef.collection('users').doc();
+        const userId = userRef.id;
 
         const gameInfo = {
             nickname: nickname,
             investments: [],
-            personal_value: 10000,
-            money_left: 10000,
+            personal_value: starting_money,
+            money_left: starting_money,
             gains: 0,
             losses: 0,
         }
 
-        this.setState({
-            net_worth: gameInfo.personal_value,
-            cash: gameInfo.money_left,
-        })
-
         userRef.set(gameInfo);
+        return userId;
     }
```

Comment:
can't you just call this method? it already exists in firestore access, why is there a duplicate?

Reply:
I think he should fix that in a separate branch and merge it over later, because the methods do have different parameters which would require some more involved changes.

Reply:
cool, make sense